### PR TITLE
Enable nullability in some GridEntry members

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in some GridEntry members.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/10170)

### CODE REVIEW

**Review by LeafShi1 — COMMENTED** [review]
LGTM!

File: src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/GridEntry.cs [hunk 1]
```diff
@@ -188,7 +193,7 @@ protected GridEntryCollection ChildCollection
 
     public int ChildCount => Children?.Count ?? 0;
 
-    public virtual GridEntryCollection Children
+    public virtual GridEntryCollection? Children
```
[the remark this answers is not in the document]
Comment (lonitra) [reply src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/GridEntry.cs:196]: Actually sorry I misread `CreateChildren()`, we cannot do the above simplification, but we can still make this not nullable. In the case `_children is null && Disposed` we can create empty `GridEntryCollection()`

[the remark this answers is not in the document]
Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/GridEntry.cs]: I added a separate commit. Is this what you had in mind?

**Review by lonitra — APPROVED** [review]
🚀

**Review by lonitra — APPROVED** [review]

**Pull request merged**